scarlet/observation.py

```diff
@@ -155,28 +152,69 @@ def match(self, model_frame, diff_kernels=None, convolution="fft"):
         self._slices_for_model = slices[1]
 
         # construct diff kernels
-        if diff_kernels is None:
-            self._diff_kernels = None
-            if self.frame.psf is not model_frame.psf:
-                assert self.frame.psf is not None and model_frame.psf is not None
-                psf = fft.Fourier(self.frame.psf.get_model().astype(model_frame.dtype))
-                model_psf = fft.Fourier(
-                    model_frame.psf.get_model().astype(model_frame.dtype)
-                )
-                self._diff_kernels = fft.match_psfs(psf, model_psf)
-        else:
-            if not isinstance(diff_kernels, fft.Fourier):
-                diff_kernels = fft.Fourier(diff_kernels)
-            self._diff_kernels = diff_kernels
+        if self.frame.psf is not model_frame.psf:
+            assert self.frame.psf is not None and model_frame.psf is not None
+            psf = fft.Fourier(self.frame.psf.get_model().astype(model_frame.dtype))
+            model_psf = fft.Fourier(
+                model_frame.psf.get_model().astype(model_frame.dtype)
+            )
+            self._diff_kernels = fft.match_psfs(psf, model_psf)
 
-        # initialize the filter window
-        assert convolution in [
+        assert convolution_type in [
             "real",
             "fft",
         ], "`convolution` must be either 'real' or 'fft'"
-        self.convolution = convolution
+        self._convolution_type = convolution_type
         return self
 
+    @property
+    def noise_rms(self):
+        import numpy.ma as ma
+
+        return 1 / np.sqrt(ma.masked_equal(self.weights, 0))
+
+    @property
+    def prerender_images(self):
+        if hasattr(self, "_prerender_images"):
+            return self._prerender_images
+        if self._diff_kernels is None:
+            return self.images
+        else:
+            # construct deconvolved image for detection:
+            # divide Fourier transform of images by Fourier transform of diff kernel
+            self._prerender_images = fft._kspace_operation(
+                fft.Fourier(self.images),
+                self._diff_kernels,
+                3,
+                fft.operator.truediv,
+                self.images.shape,
+                axes=(-2, -1),
+            ).image  # then get the image from Fourier
+            return self._prerender_images
+
+    @property
+    def prerender_sigma(self):
+        if hasattr(self, "_prerender_sigma"):
+            return self._prerender_sigma
+        if self._diff_kernels is None:
+            return np.mean(self.noise_rms, axis=(1, 2))
+        else:
+            # deconvolve noise field to estimate its noise level
```

Comment:
Because if `self._diff_kernels` is None, there is no PSF in the frame

Reply:
Yup, that what the if statement is for. My comment is about what below the comment.

Reply:
I see. I'm not familiar with the equation above, can you say how/where it is derived?

Reply:
Quadratic sum of noise. if you take a uniform noise map and  convolve  it by a psf, the std of the result is the initial nosie times the quadratic sum of the psf pixels. So just revert the equation and you have the amplitude of dedconvolved noise.

Reply:
Neat, that's just error propagation. How did I not know this?

We have non-uniform noise in general, but the equation can be generalized.

Reply:
Yup! In most of our examples, we consider that noise is uniform (though it is strictly not). If the noise is not unform we are back to solving a deconvolution problem though since the noise of a convolved noisy image is has for amplitude the map of the rms convolved by the square of the psf and taken to the root (I think). I think for the purpose of initialisation, assuming the noise is uniform will be fine.

Reply:
This also seems to be more stable than the current code! Nice!